### PR TITLE
[Merged by Bors] - perf(test/observe): set low maxHeartbeats for test

### PR DESCRIPTION
**Pull request by jcommelin**
---

See also: leanprover/std4#642

Zulip discussion: https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/exact.3F.20failure/near/421263031

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: test/observe.lean [hunk 1]
```diff
@@ -4,14 +4,19 @@ import Mathlib.Data.Nat.Prime
...
   have prime : p.Prime := by
-    apply minFac_prime
+    -- we write this in a funny way,
+    -- so that we can test that `observe` (aka `apply?`)
+    -- finds `minFac_prime` in time
     observe : n.factorial > 0
-    linarith
+    have : N ≠ 1 := by linarith
+    observe : p.Prime
+    assumption
```
Comment (semorrison) [inline test/observe.lean:19]: I don't understand this change, what is going on?

Comment (jcommelin) [reply test/observe.lean:19]: I want a way to test that `apply?` finds `minFac_prime` (fast). This is my best approximation of testing that. Ideally we should test that it appears amongst the first 10 suggestions.

Comment (semorrison) [reply test/observe.lean:19]: Could you explain *how* it tests that? I still find this completely inscrutable. :-)

Comment (semorrison) [reply test/observe.lean:19]: Can you just wrap `observe` or `apply?` with a `show_term`? You could even wrap that with a `#guard_msgs` (maybe we have a tactic version?).

Comment (jcommelin) [reply test/observe.lean:19]: Aha, I guess we would need a tactic version of `guard_msgs`... but I wouldn't know how to make this precise.

Comment (semorrison) [reply test/observe.lean:19]: I *still* do not understand why you care about this. I would much rather this file contains a nice easy proof than an inscrutable test.

File: test/observe.lean [hunk 2]
```diff
@@ -4,14 +4,19 @@ import Mathlib.Data.Nat.Prime
 
 open Nat
 
+set_option maxHeartbeats 10000 in
```
Comment (jcommelin) [inline test/observe.lean:7]:
```suggestion
set_option maxHeartbeats 4000 in
```

File: test/observe.lean [hunk 3]
```diff
@@ -4,14 +4,19 @@ import Mathlib.Data.Nat.Prime
...
 theorem euclid (n : ℕ) : ∃ N, n < N ∧ N.Prime := by
   let N := n.factorial + 1
   let p := minFac N
   use p
   have prime : p.Prime := by
-    apply minFac_prime
+    -- we write this in a funny way,
+    -- so that we can test that `observe` (aka `apply?`)
+    -- finds `minFac_prime` in time
```
Comment (jcommelin) [inline test/observe.lean:15]:
```suggestion
    apply minFac_prime
```

File: test/observe.lean [hunk 4]
```diff
@@ -4,14 +4,19 @@ import Mathlib.Data.Nat.Prime
...
     observe : n.factorial > 0
-    linarith
+    have : N ≠ 1 := by linarith
+    observe : p.Prime
+    assumption
```
Comment (jcommelin) [inline test/observe.lean:19]:
```suggestion
    linarith
```

File: test/observe.lean (hunk 1 above)
Comment (jcommelin) [reply test/observe.lean]: I've reverted those changes.

File: test/observe.lean [hunk 5]
```diff
@@ -3,6 +3,7 @@ import Mathlib.Data.Nat.Prime
 
 open Nat
 
+set_option maxHeartbeats 4000 in
```
Comment (jcommelin) [inline test/observe.lean:6]:
```suggestion
set_option maxHeartbeats 6000 in
```

**Review by grunweg — APPROVED** [review]

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**